libsel4debug/src/backtrace.c

```diff
@@ -4,12 +4,18 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 #include <autoconf.h>
+#include <sel4/types.h>
+
+/* Don't instrument seL4_GetIPCBuffer so we don't recurse.
+ * This definition has to come before the definition that is included
+ * for the attribute to have effect.
+ */
+LIBSEL4_INLINE_FUNC seL4_IPCBuffer *seL4_GetIPCBuffer(void) __attribute__((no_instrument_function));
```

Comment:
Why can't we drop the prototype here completely and just add the attribute in kernel/libsel4/include/sel4/functions.h where the implementation is?

Reply:
It's specific to the file.  The functions __cyg_profile_func_enter and __cyg_profile_func_exit call seL4_GetIPCBuffer and need to ensure that the compiler hasn't instrumented it.